File: game.py
from btagsil.repl import repl_once
from btagsil.world import init_world

def main() -> None:
    world = init_world()
    while True:
        print("==> ", sep="", end="")
        repl_once(world, input().lower())

if __name__ == "__main__":
    main()

File: src/main/scala/btagsil/Game.scala
package btagsil

import scala.io.StdIn.readLine

def mainLoop(world: World): Unit = {
    print("ScB >> ")
    val input = readLine().split(" ").filter(s => s.nonEmpty).toList

    replOnce(world, input)

    mainLoop(world)
}

object Game extends App {
    mainLoop(initWorld())
}